tests/oracle/oracle_tests.cpp

```diff
@@ -1536,44 +1540,21 @@ class test_context :public test_context_base
     }
 };
 
-int main(int argc, char** argv)
+namespace soci
+{
+namespace tests
 {
-#ifdef _MSC_VER
-    // Redirect errors, unrecoverable problems, and assert() failures to STDERR,
-    // instead of debug message window.
-    // This hack is required to run assert()-driven tests by Buildbot.
-    // NOTE: Comment this 2 lines for debugging with Visual C++ debugger to catch assertions inside.
-    _CrtSetReportMode(_CRT_ERROR, _CRTDBG_MODE_FILE);
-    _CrtSetReportFile(_CRT_ERROR, _CRTDBG_FILE_STDERR);
-#endif //_MSC_VER
-
-    if (argc >= 2)
-    {
-        connectString = argv[1];
-
-        // Replace the connect string with the process name to ensure that
-        // CATCH uses the correct name in its messages.
-        argv[1] = argv[0];
-
-        argc--;
-        argv++;
-    }
-    else
-    {
-        std::cout << "usage: " << argv[0]
-            << " connectstring [test-arguments...]\n"
-            << "example: " << argv[0]
-            << " \'service=orcl user=scott password=tiger\'\n";
-        std::exit(1);
-    }
 
-    if (!std::getenv("ORACLE_HOME"))
-    {
-        std::cerr << "ORACLE_HOME environment variable must be defined for Oracle tests.\n";
-        std::exit(1);
-    }
```

Comment:
This seems to have lost, but it would be nice to preserve it — maybe just move to `create_backend_factory()`?